src/components/RoiCalculatorModal/index.tsx

```diff
@@ -0,0 +1,493 @@
+import React, { useState, useRef, useEffect } from 'react'
+import styled from 'styled-components'
+import {
+  Modal,
+  Text,
+  LinkExternal,
+  Button,
+  Flex,
+  Box,
+  Grid,
+  ButtonMenu,
+  Checkbox,
+  PencilIcon,
+  BalanceInput,
+  HelpIcon,
+  CheckmarkIcon,
+  ButtonMenuItem,
+  Input,
+  useTooltip,
+} from '@pancakeswap/uikit'
+import BigNumber from 'bignumber.js'
+import { useTranslation } from 'contexts/Localization'
+import { getBalanceNumber } from 'utils/formatBalance'
+import { getInterestBreakdown, getPrincipalForInterest, getApy, getRoi } from 'utils/compoundApyHelpers'
+import useActiveWeb3React from 'hooks/useActiveWeb3React'
+
+interface RoiCalculatorModalProps {
+  onDismiss?: () => void
+  earningTokenPrice: number
+  apr: number
+  displayApr?: string
+  linkLabel: string
+  linkHref: string
+  stakingTokenBalance: BigNumber
+  stakingTokenSymbol: string
+  stakingTokenPrice: number
+  earningTokenSymbol?: string
+  multiplier?: string
+  roundingDecimals?: number
+  autoCompoundFrequency?: number
+  performanceFee?: number
+  isFarm?: boolean
+}
+
+const TopPart = styled(Box)`
+  background: ${({ theme }) => theme.colors.gradients.bubblegum};
+  padding: 24px;
+`
+
+const Footer = styled(Flex)`
+  background: ${({ theme }) => theme.colors.dropdown};
+`
+
+const StyledModal = styled(Modal)`
+  & > :nth-child(2) {
+    padding: 0;
+  }
+`
+
+const ScrollableContainer = styled.div`
+  max-height: 500px;
+  overflow-y: scroll;
+  ${({ theme }) => theme.mediaQueries.sm} {
+    max-height: none;
+  }
+`
+
+const FullWidthButtonMenu = styled(ButtonMenu)<{ disabled?: boolean }>`
+  width: 100%;
+
+  & > button {
+    width: 100%;
+  }
+
+  opacity: ${({ disabled }) => (disabled ? 0.5 : 1)};
+`
+
+const RoiInputContainer = styled(Flex)`
+  position: relative;
+  & > input {
+    padding-left: 28px;
+  }
+  &:before {
+    position: absolute;
+    content: '$';
+    color: ${({ theme }) => theme.colors.textSubtle};
+    left: 16px;
+    top: 8px;
+  }
+`
+
+const InvestmentInputContainer = styled.div`
+  position: relative;
+
+  & input {
+    padding-right: 34px;
+  }
+
+  &:before {
+    position: absolute;
+    content: 'USD';
+    color: ${({ theme }) => theme.colors.textSubtle};
+    right: 16px;
+    top: 21px;
+  }
+`
+
+const BulletList = styled.ul`
+  list-style-type: none;
+  margin: 0;
+  padding: 0;
+
+  li {
+    margin: 0;
+    padding: 0;
+  }
+
+  li::before {
+    content: '•';
+    margin-right: 4px;
+  }
+
+  li::marker {
+    font-size: 12px;
+    color: ${({ theme }) => theme.colors.textSubtle};
+  }
+`
+
+const compoundingIndexToFrequency = {
+  0: 2,
+  1: 1,
+  2: 0.142857142, // once every 7 days
+  3: 0.033333333, // once every 30 days
+}
+
+const getRoiData = ({ investment, interest, tokenPrice, isEditingRoi, expectedRoi, showBasedOneExpectedRoi }) => {
```

Comment:
`showBasedOneExpectedRoi` I think this might be a typo?

Reply:
fixed in ada53abe3d13902799f0f00433306f778310432d